flytekit/extend/backend/base_agent.py

```diff
@@ -211,5 +211,14 @@ def signal_handler(
         signum: int,
         frame: FrameType,
     ) -> typing.Any:
-        agent.delete(context, resource_meta)
+        if agent.asynchronous:
+            try:
+                new_loop = asyncio.new_event_loop()
+                asyncio.set_event_loop(new_loop)
```

Comment:
How did you test it? you shouldn't be able to run the new event loop while another loop is running

Reply:
I tested it in the wrong way, I will improve it no, thanks a lot!